include/ozo/connection.h

```diff
@@ -20,6 +20,62 @@ using no_statistics = decltype(hana::make_map());
 template <typename, typename = std::void_t<>>
 struct is_connection : std::false_type {};
 
+template <typename T, typename = std::void_t<>>
+struct get_connection_oid_map_impl {
+    template <typename Conn>
+    static auto apply(Conn&& c) -> decltype(c.oid_map()) { return c.oid_map();}
+    template <typename Conn>
+    static auto apply(Conn&& c) -> decltype((c.oid_map_)) { return c.oid_map_;}
```

Comment:
Why do you need additional parentheses?

Reply:
To return by reference, not by value - it works like decltype(auto). There is no other way to return by reference using decltype in SFINAE manner.